pandas/tests/indexes/test_base.py

```diff
@@ -436,6 +436,21 @@ def test_constructor_empty(self):
         assert isinstance(empty, MultiIndex)
         assert not len(empty)
 
+    def test_constructor_nonhashable_name(self):
+        # GH 20527
+        idx = self.create_index()
+        name = ['0']
+        message = "Index.name must be a hashable type"
+        tm.assert_raises_regex(TypeError, message, name=name)
+
```

Comment:
add this - it will test with all of the indices types (this is confusing atm as we have multiple way of running index creation: fixture, a function, and as a class property); other's PR's are cleaning this up.

```
diff --git a/pandas/tests/indexes/test_base.py b/pandas/tests/indexes/test_base.py
index 4265c4b56..ad7f0d7aa 100644
--- a/pandas/tests/indexes/test_base.py
+++ b/pandas/tests/indexes/test_base.py
@@ -436,9 +436,12 @@ class TestIndex(Base):
         assert isinstance(empty, MultiIndex)
         assert not len(empty)
 
-    def test_constructor_nonhashable_name(self):
+    def test_constructor_nonhashable_name(self, indices):
         # GH 20527
-        idx = self.create_index()
+
+        if isinstance(indices, MultiIndex):
+            pytest.skip("multiindex handled in test_multiindex.py")
+
         name = ['0']
         message = "Index.name must be a hashable type"
         tm.assert_raises_regex(TypeError, message, name=name)
@@ -446,10 +449,10 @@ class TestIndex(Base):
         # With .rename()
         renamed = [['1']]
         tm.assert_raises_regex(TypeError, message,
-                               idx.rename, name=renamed)
+                               indices.rename, name=renamed)
         # With .set_names()
         tm.assert_raises_regex(TypeError, message,
-                               idx.set_names, names=renamed)
+                               indices.set_names, names=renamed)
 
     def test_view_with_args(self):
```

Reply:
I pushed this to your branch

Reply:
@jreback Thanks. Can we do the same for MI (i.e. through fixture) ?